### PR TITLE
engine: mtask wire in derived context

### PR DESCRIPTION
**Pull request by adnxn**
### Summary

Wire in `context.Context` where possible for `mtask.waitEvent()` calls, else use `<-chan struct{}` signalling channel instead of `<-chan bool`
### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

This contribution is under the terms of the Apache 2.0 License:

### CODE REVIEW

File: agent/engine/common_test.go [hunk 1]
```diff
@@ -184,20 +184,15 @@ func addTaskToEngine(t *testing.T,
 	createStartEventsReported sync.WaitGroup) {
 	// steadyStateCheckWait is used to force the test to wait until the steady-state check
 	// has been invoked at least once
-	steadyStateCheckWait := sync.WaitGroup{}
```
Comment (aaithal) [inline agent/engine/common_test.go]: why was this removed? steadyState check should not have any bearing on wiring contexts to `waitEvent` method, right?

Comment (adnxn) [reply agent/engine/common_test.go]: I removed this because further down we do a `steadyStateCheckWait.Done()` for a `mocktime.EXPECT()...`, since this mocktime maps to this removed [code](https://github.com/aws/amazon-ecs-agent/pull/1226/files/e42e6b13f0813b3d5ce6f8115ef6751b304d46ba#diff-20346f7af6202828de353f0137b6e43aL264) in the `waitSteady()` method. I don't *think* we're losing any coverage by getting rid of this here. Let me know if you think I'm missing something.

File: agent/engine/task_manager.go [hunk 2]
```diff
@@ -238,12 +241,16 @@ func (mtask *managedTask) waitForHostResources() {
 
 	seelog.Infof("Managed task [%s]: waiting for any previous stops to complete. Sequence number: %d",
 		mtask.Arn, mtask.StartSequenceNumber)
-	othersStopped := make(chan bool, 1)
+
+	othersStoppedCtx, othersStoppedCancel := context.WithCancel(mtask.ctx)
```
[the remark this answers is not in the document]
Comment (adnxn) [reply agent/engine/task_manager.go]: makes sense, it's a small block of code.

File: agent/engine/common_test.go (hunk 1 above)
Comment (richardpen) [reply agent/engine/common_test.go]: Since we are getting rid of the `mocktime` for `waitSteady`, how are making sure the `waitSteady` is called? Also can the line 194 be removed?

Comment (adnxn) [reply agent/engine/common_test.go]: > Also can the line 194 be removed?

We can remove it.

> how are making sure the waitSteady is called? 

Are we losing anything by not ensuring that waitSteady is being called? Since it is a private method and an implementation detail, I'm not sure we need to be explicitly checking if waitSteady is called. Please let me know what you think though.
